demo/Demo/Tooltip.elm

```diff
@@ -0,0 +1,222 @@
+module Demo.Tooltip exposing (..)
+
+import Platform.Cmd exposing (Cmd, none)
+import Html exposing (..)
+import Html.Attributes as Html exposing (..)
+
+import Material.Tooltip as Tooltip
+import Material
+
+import Demo.Page as Page
+
+import Material.Grid as Grid
+import Demo.Code as Code
+import Markdown
+
+import Material.Button as Button
+import Material.Icon as Icon
+
+-- MODEL
+
+type alias Model =
+  { mdl : Material.Model
+  , nopart : Tooltip.Model
+  }
+
+
+model : Model
+model =
+  { mdl = Material.model
+  , nopart = Tooltip.defaultModel
+  }
+
+
+-- ACTION, UPDATE
+type Msg
+  = NoOp
+  | TooltipMsg Tooltip.Msg
+  | Mdl Material.Msg
+
+
+update : Msg -> Model -> (Model, Cmd Msg)
+update action model =
+  case action of
+    NoOp ->
+        (model, Cmd.none)
+
+    TooltipMsg msg' ->
+      let
+        updated = fst <| (Tooltip.update msg' model.nopart)
+      in
+        ({ model | nopart = updated }, Cmd.none)
+
+    Mdl action' ->
+      Material.update Mdl action' model
+
+
+code : String -> Html a
+code str =
+  div
+    [ style [("overflow", "hidden")] ]
+    [ Markdown.toHtml [] <| "```elm\n" ++ Code.trim str ++ "\n```" ]
+
+-- VIEW
+
+
+demoTooltip : (Html a, String) -> Grid.Cell a
+demoTooltip (tooltip, description) =
+  Grid.cell
+    [Grid.size Grid.All 4]
+    [ div [style [("text-align", "center")]]
+        [tooltip]
+    , code description
+    ]
+
+view : Model -> Html Msg
+view model  =
+  [ div
+      []
+      [ Html.p [] [text "Example use:"]
+      , Grid.grid []
+          [Grid.cell [Grid.size Grid.All 12]
+             [code """
+                    import Material.Tooltip as Tooltip
+                    import Material.Icon as Icon
+
+                    -- Note the index in both the Render as well as the mouse event handlers
+
+                    tooltip : Model -> Html Msg
+                    tooltip model =
+                      div []
+                        [ Icon.view "add" [ Tooltip.attach Mdl [0] ]
+                        , Tooltip.render Mdl [0] model.mdl
+                            [Tooltip.default]
+                            [text "Default tooltip"]
+                        ]
+                    """]
+          ]
+
+
+      , Grid.grid []
+          [ demoTooltip
+            (div []
+               [ Icon.view "add"
+                   [ Tooltip.attach Mdl [0] ]
+
+               , Tooltip.render Mdl [0] model.mdl
+                   [Tooltip.default]
+                   [text "Default tooltip"]
+               ]
+            , """
```

Comment:
Generally need more space between the element with the tooltip and the description below. Especially Button.